go/mysql/binlog_event_json.go

```diff
@@ -114,27 +118,32 @@ func printJSONObject(data []byte, large bool, result *bytes.Buffer) error {
 	}
 
 	// Build an array for each key.
-	keys := make([]sqltypes.Value, elementCount)
+	keys := make([]keyData, elementCount)
 	for i := 0; i < elementCount; i++ {
 		var keyOffset, keyLength int
 		keyOffset, pos = readOffsetOrSize(data, pos, large)
 		keyLength, pos = readOffsetOrSize(data, pos, false) // always 16
-		keys[i] = sqltypes.MakeTrusted(sqltypes.VarBinary, data[keyOffset:keyOffset+keyLength])
+		keys[i] = keyData{keyOffset, keyLength}
 	}
 
 	// Now read each value, and output them.  The value entry is
 	// always one byte (the type), and then 2 or 4 bytes
 	// (depending on the large flag). If the value fits in the number of bytes,
 	// then it is inlined. This is always the case for Literal (one byte),
 	// and {,u}int16. For {u}int32, it depends if we're large or not.
-	result.WriteString("JSON_OBJECT(")
+	result.WriteString("{")
 	for i := 0; i < elementCount; i++ {
 		// First print the key value.
 		if i > 0 {
 			result.WriteByte(',')
 		}
-		keys[i].EncodeSQL(result)
-		result.WriteByte(',')
+		keydata := keys[i]
+		//(data[keyOffset:keyOffset+keyLength])
+		result.WriteString("\"")
+		// FIXME(alainjobart): escape reserved characters
+		result.Write(data[keydata.offset : keydata.offset+keydata.length])
```

Comment:
We have to test for `"` and other characters that json requires you to escape. I think they'll break. Same comments below for `printJSONString`.

Reply:
Since this is already a problem with the existing implementation, can we defer the handling of reserved characters? As it stands today, basic resharding with vreplication fails until this PR is merged.